src/stlink-lib/common_flash.c

```diff
@@ -400,22 +423,32 @@ static void unlock_flash(stlink_t *sl) {
     key_reg = FLASH_F4_KEYR;
   } else if (sl->flash_type == STM32_FLASH_TYPE_F7) {
     key_reg = FLASH_F7_KEYR;
-  } else if (sl->flash_type == STM32_FLASH_TYPE_L0_L1) {
-    key_reg = get_stm32l0_flash_base(sl) + FLASH_PEKEYR_OFF;
-    flash_key1 = FLASH_L0_PEKEY1;
-    flash_key2 = FLASH_L0_PEKEY2;
-  } else if (sl->flash_type == STM32_FLASH_TYPE_L4) {
-    key_reg = STM32L4_FLASH_KEYR;
   } else if (sl->flash_type == STM32_FLASH_TYPE_G0 ||
              sl->flash_type == STM32_FLASH_TYPE_G4) {
     key_reg = STM32Gx_FLASH_KEYR;
-  } else if (sl->flash_type == STM32_FLASH_TYPE_WB_WL) {
-    key_reg = STM32WB_FLASH_KEYR;
   } else if (sl->flash_type == STM32_FLASH_TYPE_H7) {
     key_reg = FLASH_H7_KEYR1;
     if (sl->chip_flags & CHIP_F_HAS_DUAL_BANK) {
       key2_reg = FLASH_H7_KEYR2;
     }
+  } else if (sl->flash_type == STM32_FLASH_TYPE_L0_L1) {
+    key_reg = get_stm32l0_flash_base(sl) + FLASH_PEKEYR_OFF;
+    flash_key1 = FLASH_L0_PEKEY1;
+    flash_key2 = FLASH_L0_PEKEY2;
+  } else if (sl->flash_type == STM32_FLASH_TYPE_L4) {
+    key_reg = STM32L4_FLASH_KEYR;
+  } else if (sl->flash_type == STM32_FLASH_TYPE_L5_U5) {
+    // Set voltage scaling to range 0 to perform flash operations (RM0438 p. 183)
+    uint32_t mask = (0b11 << STM32L5_PWR_CR1_VOS);
+    uint32_t val;
+    stlink_read_debug32(sl, STM32L5_PWR_CR1, &val);
```

Comment:
It is better to check the return code, otherwise you can spoil the value of the PWR_CR1 register

Reply:
Not sure on how to address this one exactly...

Reply:
Ah, I see... I thought you meant something more complex... Easy then. :+1: